packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1483,6 +1483,10 @@ class RenderEditable extends RenderBox {
     // When long-pressing past the end of the text, we want a collapsed cursor.
     if (position.offset >= word.end)
       return TextSelection.fromPosition(position);
+    // If text is obscured, the entire sentence should be treated as one word.
+    if (obscureText) {
+      return TextSelection(baseOffset: 0, extentOffset: text.toPlainText().length);
```

Comment:
Is this the right way to get the full length of the text? @GaryQian may know.

Reply:
Generally, this will indeed produce the length of the text content of the `text` object. However, to properly and explicitly handle things like PlaceholderSpan/WidgetSpans, you may have to look deeper.

toPlainText will return the concatenated contents for TextSpan. When encountering PlaceholderSpans, the placeholder span will be ignored unless includePlaceholders is true, in which case a 0xFFFC character of length 1 will be included. Whether or not you want this replacement character is a policy choice. In addition, if another text widget is nested using a WidgetSpan, the length of the nested contents will not be included.

Reply:
Although this looks like it is for obscuring text in a password like environment, which I would expect would not be using PlaceholderSpans. It is hard to say what you would expect if one were to, say, embed all of flutter gallery inline within a password field. How many character lengths is a gallery equivalent to?

Just setting the includePlaceholders flag to true seems to me like a reasonable compromise, where all placeholders/inline widgets will be represented as one character.

Reply:
the includePlaceholders is defaulted to true. As of now, there is no way to use any other inlinespan other than textspan in RenderEditable. There is no plan to add it in until we figure out how do we serialize them. We only need to worry about text span for now.